airbyte-webapp/src/pages/ConnectorBuilderPage/LandingPage.tsx

```diff
@@ -0,0 +1,199 @@
+  }: {
+    hideLandingPage: () => void;
+    switchToUI: () => void;
+    switchToYaml: () => void;
+  }) => {
+    const fileInputRef = useRef<HTMLInputElement>(null);
+    const { registerNotification, unregisterNotificationById } = useNotificationService();
+    const { setJsonManifest, setYamlIsValid } = useConnectorBuilderFormState();
+    const { convertToBuilderFormValues } = useManifestToBuilderForm();
+    const { setValues } = useFormikContext<BuilderFormValues>();
+    const [importYamlLoading, setImportYamlLoading] = useState(false);
+
+    const handleYamlUpload = useCallback(
+      async (yaml: string, fileName?: string) => {
+        try {
+          let json;
+          try {
+            json = load(yaml) as ConnectorManifest;
+          } catch (e) {
+            if (e instanceof YAMLException) {
+              registerNotification({
+                id: YAML_UPLOAD_ERROR_ID,
+                text: (
+                  <FormattedMessage
+                    id={YAML_UPLOAD_ERROR_ID}
+                    values={{
+                      reason: e.reason,
+                      line: e.mark.line,
+                    }}
+                  />
+                ),
+                type: ToastType.ERROR,
+              });
+            }
+            return;
+          }
+          setYamlIsValid(true);
+
+          let convertedFormValues;
+          try {
+            convertedFormValues = await convertToBuilderFormValues(json, DEFAULT_BUILDER_FORM_VALUES);
+          } catch (e) {
+            switchToYaml();
+            setJsonManifest(json);
+            hideLandingPage();
+            return;
+          }
+
+          if (fileName) {
+            convertedFormValues.global.connectorName = startCase(fileName.split(".")[0]);
+          }
+          setValues(convertedFormValues);
+          hideLandingPage();
+        } finally {
+          if (fileInputRef.current) {
+            fileInputRef.current.value = "";
+          }
+          setImportYamlLoading(false);
+        }
+      },
+      [
+        convertToBuilderFormValues,
+        registerNotification,
+        setJsonManifest,
+        hideLandingPage,
+        setValues,
+        setYamlIsValid,
+        switchToYaml,
+      ]
+    );
+
+    return (
+      <FlexContainer direction="column" alignItems="center" gap="2xl">
+        <FlexContainer direction="column" gap="md" alignItems="center" className={styles.titleContainer}>
+          <AirbyteLogo />
+          <Heading as="h1" size="lg" className={styles.title}>
+            <FormattedMessage id="connectorBuilder.landingPage.title" />
+          </Heading>
+        </FlexContainer>
+        <Heading as="h1" size="lg">
+          <FormattedMessage id="connectorBuilder.landingPage.prompt" />
+        </Heading>
+        <FlexContainer direction="row" gap="2xl">
+          <input
+            type="file"
+            accept=".yml,.yaml"
+            ref={fileInputRef}
+            style={{ display: "none" }}
```

Comment:
nit: the prop `hidden` should work as well here and is slightly more "correct" (not a big deal though)